presto-tests/src/main/java/com/facebook/presto/tests/datatype/DataType.java

```diff
@@ -34,6 +44,46 @@
     private Function<T, String> toLiteral;
     private Function<T, ?> toPrestoQueryResult;
 
+    public static DataType<Boolean> booleanDataType()
+    {
+        return dataType("boolean", BooleanType.BOOLEAN);
+    }
+
+    public static DataType<Long> bigintDataType()
+    {
+        return dataType("bigint", BigintType.BIGINT);
+    }
+
+    public static DataType<Integer> integerDataType()
+    {
+        return dataType("integer", IntegerType.INTEGER);
+    }
+
+    public static DataType<Short> smallintDataType()
+    {
+        return dataType("smallint", SmallintType.SMALLINT);
+    }
+
+    public static DataType<Byte> tinyintDataType()
+    {
+        return dataType("tinyint", TinyintType.TINYINT);
+    }
+
+    public static DataType<Double> doubleDataType()
+    {
+        return dataType("double", DoubleType.DOUBLE);
+    }
+
+    public static DataType<Float> realDataType()
+    {
+        return dataType("real", RealType.REAL);
+    }
+
+    public static DataType<SqlVarbinary> varbinaryDataType()
```

Comment:
Why not `DataType<byte[]>`? Creation of test data would be simpler.

Reply:
For this to work you'd need:

* #9471
* using `ByteBuffer::wrap` as `toPrestoQueryResult` rather than `Function.identity()`

(if you choose to go with `SqlVarbinary` anyway, you need similar changes.. )

Reply:
`byte[]` doesn't work (as you found) because arrays aren't comparable, so I used `SqlVarBinary`. I didn't realize `ByteBuffer` was available.

Reply:
`byte[]` won't compare but it doesn't have to. You can `DataType<byte[]>` (this dictates that constants in test are `byte[]` which is easy to instantiate) and:

* have non-identity `DataType#toPrestoQueryResult` converting to a comparable type, the same as returned by `TestingPrestoClient#convertToRowValue` (ie. be `SqlVarbinary` after your change) OR
* leave  `TestingPrestoClient#convertToRowValue` unchanged (it would return `byte[]`) and then, in `DataTypeTest#execute` replace `assertEquals(row.getFields(), expectedResults)` with the code below. This will let TestNG realize it's an array and do value-based comparison.
```
for (int i = 0; i < row.getFieldCount(); i++) {
    assertEquals(row.getField(i), expectedResults.get(i));
}
```